### DISCUSSION TITLE
Test real BNG

### DISCUSSION THREAD

**Discussion by broonu** [rtbrick/bngblaster#172] [Q&A] [answered]
Hello,

Can I use bngblaster to test a real BNG?
I read the Quickstart and Access Protocols/PPPoE and the configuration is to emulate both clients and servers.

**Comment by hannesgredler** [comment, marked as answer]
Yes, BNG Blaster is used every day for testing against real BNGs - give it a try !

**Comment by GIC-de** [comment]
The example in the Quickstart Guide emulates client and server to allow users to easily get in touch with the software, without the need of having a real network. But as Hannes mention, it is build and used for testing real BNG and other types of routers/network functions.

**Comment by broonu** [comment]
Thank you guys!

Is there a documentation of how to do this?

**Reply by hannesgredler** [reply]
Please check https://rtbrick.github.io/bngblaster/ for our documentation

**Reply by broonu** [reply]
Thank you!

I read this doc and in Access Protocols/PPPoE I found:

`{
    "interfaces": {
        "network": {
            "interface": "eth2",
            "address": "10.0.0.1/24",
            "gateway": "10.0.0.2",
            "address-ipv6": "fc66:1337:7331::1/64",
            "gateway-ipv6": "fc66:1337:7331::2"
        },
        "access": [
            {
                "interface": "eth1",
                "type": "pppoe",`
                
I have a Linux test machine with ens224 used for PPPoE with PPPD.
In this DOC I need a "network interface" and a "access interface", I need to use ens224 as "network" and create another 'access" interface?

**Reply by GIC-de** [reply]
The network interface is optional but required to generate traffic. The idea is simple, the access interface emulates the access protocols like PPPoE, ... and the network interface can be used as destination for traffic. This allows to setup sessions and verify them using traffic between session and network interface. The network interface can be connected directly to BNG or somewhere in your core. But emulating PPPoE sessions even works without network interface, just without bidirectional traffic.

**Reply by broonu** [reply]
> The network interface is optional but required to generate traffic. The idea is simple, the access interface emulates the access protocols like PPPoE, ... and the network interface can be used as destination for traffic. This allows to setup sessions and verify them using traffic between session and network interface. The network interface can be connected directly to BNG or somewhere in your core. But emulating PPPoE sessions even works without network interface, just without bidirectional traffic.

Thank you.
I'm trying with a basic configuration I found in a DENOG vídeo, like this:

`{
  "__comment__": "PPPoE",
  "interfaces": {
    "access": [
      {
        "__comment__": "PPPoE",
        "interface": "ens224",
        "type": "pppoe",
        "username": "my_user",
        "password": "my_pass"
      }
    ]
  }
}
`
Should it work to connect in my BNG through interface ens224?

**Reply by GIC-de** [reply]
You can find examples even here:
https://github.com/rtbrick/bngblaster/tree/main/examples

**Reply by broonu** [reply]
> You can find examples even here: https://github.com/rtbrick/bngblaster/tree/main/examples

Thank you, but this example is to emulate client and server.
I can't find an example using a real BNG, just emulated.

**Comment by GIC-de** [comment]
No, those examples emulate client only for testing with real BNG.

broonu ***@***.***> schrieb am Mi. 26. Apr. 2023 um 19:36:

> You can find examples even here:
> https://github.com/rtbrick/bngblaster/tree/main/examples
>
> Thank you, but this example is to emulate client and server.
> I can't find an example using a real BNG, just emulated.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rtbrick/bngblaster/discussions/172#discussioncomment-5734558>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABONJOFKVFEPR6RASTHY2BLXDFMLXANCNFSM6AAAAAAXLYBDU4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

-- 
NOTICE TO
RECIPIENT This e-mail message and any attachments are 
confidential and may be
privileged. If you received this e-mail in error, 
any review, use,
dissemination, distribution, or copying of this e-mail is 
strictly
prohibited. Please notify us immediately of the error by return 
e-mail and
please delete this message from your system. For more 
information about Rtbrick, please visit us at www.rtbrick.com 
<http://www.rtbrick.com>

**Reply by broonu** [reply]
Can you help me this config?
I'm using a basic configuration:
```
{
  "__comment__": "PPPoE",

  "interfaces": {
    "access": [
      {
       "__comment__": "PPPoE",
       "interface": "ens224",
       "type": "pppoe",
       "username": "my_user",
       "password": "my_pass"
      }
    ]
  }
}
```

The report shows 8 PADI sent, but I can't view this PADI and my BNG didn't receive it:
```
Access Interface Protocol Packet Stats:
  ARP    TX:          0 RX:          0
  PADI   TX:          8 RX:          0
  PADO   TX:          0 RX:          0
  PADR   TX:          0 RX:          0
  PADS   TX:          0 RX:          0
  PADT   TX:          1 RX:          0
  LCP    TX:          0 RX:          0
  PAP    TX:          0 RX:          0
  CHAP   TX:          0 RX:          0
  IPCP   TX:          0 RX:          0
  IP6CP  TX:          0 RX:          0
  IGMP   TX:          0 RX:          0
  ICMP   TX:          0 RX:          0
  DHCP   TX:          0 RX:          0
  DHCPv6 TX:          0 RX:          0
  ICMPv6 TX:          0 RX:          0
  IPv4 Fragmented       RX:          0
```

ens224 is the same interface I use with pppd.

**Reply by broonu** [reply]
Nevermind, it was a problem in VMware. It's working now! Thank yoi!

**Reply by jozefrebjak** [reply]
@broonu please what was a problem ? I have simillar issue. Where I triying to simulate to Cisco ASR 1001-x from VM in VMware thanks

**Reply by GIC-de** [reply]
If you start tcpdump, you do not see any traffic send by BNG Blaster, since it sends using kernel bypass. This is why the BNG Blaster supports the option to capture directly with `-P test.pcap`.

https://rtbrick.github.io/bngblaster/troubleshooting.html#pcap

In VMware, they sometimes do not accept VLAN tagged traffic from untagged interface, so you need to setup the interfaces correctly to transparently pass VLAN tags. But I'm not a VMware expert. Maybe if someone can share, I can put on our FAQ.

**Reply by jozefrebjak** [reply]
@GIC-de I make it work. I'm using DSW in VMware and I updated Security setting of Port Group like

<img width="1460" height="943" alt="CleanShot 2025-09-26 at 12 43 40" src="https://github.com/user-attachments/assets/641ee62f-56fa-4739-9a1c-754f2b760489" />

**Reply by GIC-de** [reply]
Thanks a lot.

**Reply by jozefrebjak** [reply]
My config is currenly like 

```json
{
  "interfaces": {
    "access": [
      {
        "interface": "ens192",
        "vlan-mode": "N:1",
        "type": "pppoe",
        "outer-vlan-min": 0,
        "outer-vlan-max": 0,
        "inner-vlan-min": 0,
        "inner-vlan-max": 0,
        "username": "user{session-global}@test.com",
        "password": "test"
      }
    ]
  },
  "sessions": {
    "count": 10
  },
  "pppoe": {
    "reconnect": true
  }
}
```

I'm trying to simulate how much sessions can handle Cisco ASR 1001-x to be terminated. Now I just need to somehow configure Cisco to allow generated usernames to Terminate
